src/Events/NavigateToScene.php

```diff
@@ -0,0 +1,69 @@
+<?php
+declare(strict_types=1);
+
+namespace LotGD\Core\Events;
+
+
+use Doctrine\Common\Util\Debug;
+use Doctrine\DBAL\Schema\View;
+use LotGD\Core\Exceptions\ArgumentException;
+use LotGD\Core\Models\Character;
+use LotGD\Core\Models\Scene;
+use LotGD\Core\Models\Viewpoint;
+
+
+/**
+ * NavigateToScene data container which can be used for navigational events.
+ *
+ * Fields are:
+ *  referrer    Scene|null
+ *  viewpoint   Viewpoint
+ *  scene       Scene
+ *  parameters  array
+ *  redirect    Scene|null
+ * @package LotGD\Core\Events
+ */
+class NavigateToScene extends EventContextDataContainer
```

Comment:
Let's call these objects `NavigateToSceneData` etc. I know it's long but I was confused by the verb names in your example. Having the noun attached to the end makes it consistent with the superclass and clearer when reading.